### PR TITLE
Update cppcheck and minor fixes

### PR DESCRIPTION
**Pull request by mattclarke**
### Issue

N/A

### Description of work

Updated CppCheck to be the same as the Forwarder and fixed some of the warnings.

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -35,6 +35,12 @@ elseif(NOT ${CONAN} MATCHES "DISABLE")
   MESSAGE(FATAL_ERROR "Unrecognised option for CONAN, use AUTO, MANUAL or DISABLE")
 endif()
 
+# Yes, there exists also FindGTest.
+# Yes, we do want to use this one instead:
+if(NOT GOOGLETEST_DISABLE)
```
Comment (matthew-d-jones) [inline CMakeLists.txt]: Should this use the `BUILD_TESTS` parameter you added to the readme?

File: src/HDFFile.cpp [hunk 2]
```diff
@@ -242,7 +242,8 @@ void writeAttributes(hdf5::node::Node const &Node, nlohmann::json const *Value,
 /// \param JsonValue    json value array of attribute objects.
 /// \param Logger Pointer to spdlog instance to be used for logging.
 void writeArrayOfAttributes(hdf5::node::Node const &Node,
-                            const nlohmann::json &Values, SharedLogger Logger) {
+                            const nlohmann::json &Values,
+                            SharedLogger const &Logger) {
```
Comment (matthew-d-jones) [inline src/HDFFile.cpp]: East const, west const, east const -.-
Maybe we should standardise?

Comment (mattclarke) [reply src/HDFFile.cpp]: East const for me

Comment (mattclarke) [reply src/HDFFile.cpp]: That also seems to be most prevalent.

Comment (matthew-d-jones) [reply src/HDFFile.cpp]: Ok, fine with me.
I'll do my best to break my west const habit... I think Jack also picked up west const from me. Jonas uses east const though. Will try to remember to bring it up in a meeting.

**Pull request merged**